### PR TITLE
Add support zephyr-sof for imx8ulp platform

### PR DESCRIPTION
**Pull request by TangleZ**
Add build sof with Zephyr on iMX8.ULP platform.

### CODE REVIEW

**Comment by iuliana-prodan** [comment]
LGTM. Please @TangleZ fix checkpatch issues.

@marc-hb how do we add this target in CI?
To have it here:` Zephyr / build-linux (mnfst, imx8 imx8x imx8m) (pull_request)`  or 
`Zephyr / build-linux (zmain, imx8 imx8x imx8m) (pull_request)` 
There is no rush for this, because we have dependencies on Zephyr's PR https://github.com/zephyrproject-rtos/zephyr/pull/63751 and sdk: https://github.com/zephyrproject-rtos/sdk-ng/pull/712
But just so we know what to do when all is in place :)

**Review by dbaluta — APPROVED** [review]

**Comment by marc-hb** [comment]
> @marc-hb how do we add this target in CI?

.github/workflows/zephyr.yml

> I think there is a name collision.

I think I know why, I will take a look today.

**Review by lgirdwood — APPROVED** [review]
Good stuff - Its looks actually quite simple to add a new platform using Zephyr now :)

**Comment by marc-hb** [comment]
> I think there is a name collision.

Fix submitted:
- #8441

File: app/sample.yaml [hunk 1]
```diff
@@ -16,3 +16,4 @@ tests:
       - nxp_adsp_imx8
       - nxp_adsp_imx8x
       - nxp_adsp_imx8m
+      - nxp_adsp_imx8ulp
```
Comment (marc-hb) [inline app/sample.yaml:19]: I suspect you can't do that without listing it in `platform_allow` too. Try `./scripts/twister -T ../modules/audio/sof/app/`

Strange, there should be a git conflict with #8429 merged 4 hours ago... GitHub is probably running behind, there were some outages today.

**Review by kv2019i — APPROVED** [review]
@TangleZ @dbaluta We can merge immediately or if you want to look at the sample app problem (see comment from Marc). Looks good to me!

**Comment by dbaluta** [comment]
@kv2019i merged this now. Looks like github now how to handle conflicts at rebase. 

Also, I don't think I understand what @marc-hb  says with `platform_allow`.

**Comment by marc-hb** [comment]
> Later edit: @TangleZ you need to add imx8ulp in platform_allow section:

Yes, that's it.


Never blindly search/replace without testing. I always add garbage in unknown files first. If garbage does not break anything then it means that file is not being tested. Always break things first, then fix them.

To test that file run this:

```
./zephyr/scripts/twister -T sof/app/

INFO    - 6 of 6 test configurations passed (100.00%), 0 failed, 0 errored, 0 skipped with 0 warnings in 64.87 seconds

grep platform twister-out/twister.json  | sort
            "platform":"intel_adsp_ace15_mtpm",
            "platform":"intel_adsp_ace20_lnl",
            "platform":"intel_adsp_cavs25",
            "platform":"nxp_adsp_imx8",
            "platform":"nxp_adsp_imx8m",
            "platform":"nxp_adsp_imx8x",

```

  => no imx8ulp

File: app/sample.yaml (hunk 1 above)
Comment (marc-hb) [reply app/sample.yaml:19]: There was no git conflict, the changes were far away enough from each other.

**Comment by marc-hb** [comment]
> @marc-hb how do we add this target in CI?

Actually: you don't because it does not compile yet. Not defined in Zephyr yet.

Fix for `--all` submitted in:
- #8493

**Comment by marc-hb** [comment]
> Later edit: @TangleZ you need to add imx8ulp in platform_allow section:

Finally submitted by @nashif in #8768

**Pull request merged**